File: addon_configs/a0d7b954_appdaemon/apps/controllers/poolpumpcontroller.py
import hassapi as hass

KEY_PUMP_SWITCH = "pool_pump_switch"
KEY_POOL_PUMP_BUTTON = "pool_pump_button"

#pool_pump_controller:
#  module: PoolPumpController
#  pool_pump_switch: switch.switch_7_7_pool_pump
#  pool_pump_button: binary_input.binary_sensor_7_1_pool_pump_button

# Declare Class
class PoolPumpController(hass.Hass):
    def initialize(self):
        self.listen_state(self.button_callback, self.args[KEY_POOL_PUMP_BUTTON], attribute = "state")        
        self._firstRun = True

    def __pump_pool_on(self,run):
        self.log(f"__pump_pool_on run:{run}") 
        if (run==True):
            self.turn_on(self.args[KEY_PUMP_SWITCH])
        else:
            self.turn_off(self.args[KEY_PUMP_SWITCH])

    def button_callback(self, entity, attribute, old, new, kwargs):        
        __switch_state = self.get_state(self.args[KEY_PUMP_SWITCH], attribute="state")
        self.log(f"pool button callback {entity} state: {__switch_state}, first run: {self._firstRun}")
        if (self._firstRun==False):            
            if (__switch_state=="on"):
                self.__pump_pool_on(False)
            else:
                self.__pump_pool_on(True)
        else:
            self._firstRun = False
            self.__pump_pool_on(False)
            self.log(f"first run pool button callback {entity} state: {__switch_state}")

File: netdaemon4/HomeApps/Controllers/PoolPumpController.cs
using System.Security.Cryptography.X509Certificates;
using NetDaemon.HassModel.Entities;
using NetDaemon.Extensions;

namespace HassModel.HomeApps.Controllers
{
    [NetDaemonApp]
    public class PoolPumpController
    {

        #region Constants

        public const string SOLARPUMP_MODE_ON = "On";
        public const string SOLARPUMP_MODE_OFF = "Off";
        public const string SOLARPUMP_MODE_AUTO = "Auto";
        public const string SOLARPUMP_MODE_SWIMMING = "Swimming";

        public const string pool_pump_switch = "switch.switch_7_7_pool_pump";
        public const string pool_pump_button = "binary_input.binary_sensor_7_1_pool_pump_button";
        public const string set_temperature_pump_turn_on = "";
        public const string set_temperature_pump_turn_off = "";
        public const string temperature_solar_panel_id = "sensor.temp_solar_heater";
        public const string poolpump_mode = "input_select.solar_pump_mode";

        #endregion

        private ILogger<PoolPumpController> logger;

        private bool firstRun;

        public Entity PumpSwitchEntity { get; set; }
        public Entity PoolPumpButtonEntity { get; set; }

        public Entity SensorTemperatureSolarPanel { get; set; }
    
        public PoolPumpController(IHaContext ha, ILogger<PoolPumpController> logger)
        {            
            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
            this.firstRun = true;
            PumpSwitchEntity = ha.Entity(pool_pump_switch);
            PoolPumpButtonEntity = ha.Entity(pool_pump_button);
            SensorTemperatureSolarPanel = ha.Entity(temperature_solar_panel_id);

            PoolPumpButtonEntity.StateChanges().Subscribe(OnPoolPumpButton);
            SensorTemperatureSolarPanel.StateChanges().Subscribe(OnSensorTemperatureSolarPanelChange);
            logger.LogInformation("PoolPumpController started");
        }

        public void OnSensorTemperatureSolarPanelChange(StateChange state)
        {
            logger.LogInformation("OnSensorTemperatureSolarPanelChange callback {0} state: {1}, first run: {2}"
                , SensorTemperatureSolarPanel.EntityId, SensorTemperatureSolarPanel.State, this.firstRun);
        }

        public void OnPoolPumpButton(StateChange state)
        {
            var switchStateOn = PumpSwitchEntity.IsOn();
            logger.LogInformation("pool button callback {0} state: {1}, first run: {2}", PoolPumpButtonEntity.EntityId, PumpSwitchEntity.State, this.firstRun);
            this.firstRun = false;
        }
    }
}